pkg/block/fetcher.go

```diff
@@ -401,14 +401,25 @@ func NewLabelShardedMetaFilter(relabelConfig []*relabel.Config) *LabelShardedMet
 	return &LabelShardedMetaFilter{relabelConfig: relabelConfig}
 }
 
+const blockIDLabel = "__block_id"
+
 // Filter filters out blocks that filters blocks that have no labels after relabelling.
 func (f *LabelShardedMetaFilter) Filter(metas map[ulid.ULID]*metadata.Meta, synced GaugeLabeled, _ bool) {
+	lbls := make(labels.Labels, 1)
 	for id, m := range metas {
-		if processedLabels := relabel.Process(labels.FromMap(m.Thanos.Labels), f.relabelConfig...); processedLabels != nil {
-			continue
+		lbls[0] = labels.Label{Name: blockIDLabel, Value: id.String()}
+		lbls = lbls[:1]
+		if len(lbls) < len(m.Thanos.Labels) {
+			lbls = make([]labels.Label, 0, len(m.Thanos.Labels))
```

Comment:
this logic is a bit funny I think. If `m.Thanos.Labels` has length `2`, then we throw away the block ID label we set earlier. But if it has length `1`, then we keep the block ID label and allocate a new backing array when we append that one single label. Is that what we want to do?

Reply:
Yeah it looks like in this case we wouldn't end up having the block id label I think. I can see what you were trying to do but is this really that performance sensitive that we need to safe the allocations here?

Reply:
Good point. I wanted to save allocs, will fix